pyvista/trame/jupyter.py

```diff
@@ -0,0 +1,275 @@
+    if server_proxy_enabled is None:
+        server_proxy_enabled = pyvista.global_theme.trame.server_proxy_enabled
+    if server_proxy_enabled:
+        if server_proxy_prefix is None:
+            server_proxy_prefix = pyvista.global_theme.trame.server_proxy_prefix
+        # server_proxy_prefix assumes trailing slash
+        src = (
+            f"{server_proxy_prefix if server_proxy_prefix else ''}{_server.port}/index.html{params}"
+        )
+    else:
+        src = f'{kwargs.get("protocol", "http")}://{kwargs.get("host", "localhost")}:{_server.port}/index.html{params}'
+    logger.debug(src)
+    return src
+
+
+def initialize(
+    server, plotter, mode=None, default_server_rendering=True, collapse_menu=False, **kwargs
+):
+    """Generate the UI for a given plotter."""
+    state = server.state
+    state.trame__title = UI_TITLE
+
+    viewer = get_or_create_viewer(plotter, suppress_rendering=mode == 'client')
+
+    with VAppLayout(server, template_name=plotter._id_name):
+        viewer.ui(
+            mode=mode,
+            default_server_rendering=default_server_rendering,
+            collapse_menu=collapse_menu,
+            **kwargs,
+        )
+
+    return viewer
+
+
+def show_trame(
+    plotter,
+    mode=None,
+    name=None,
+    server_proxy_enabled=None,
+    server_proxy_prefix=None,
+    collapse_menu=False,
+    default_server_rendering=True,
+    return_viewer=False,
+    **kwargs,
+):
+    """Run and display the trame application in jupyter's event loop.
+
+    plotter : pyvista.BasePlotter
+        The PyVista plotter to show.
+
+    mode : str, optional
+        The UI view mode. This can be set on the global theme. Options are:
+            * ``'trame'``: Uses a view that can switch between client and server
+              rendering modes.
+            * ``'server'``: Uses a view that is purely server rendering.
+            * ``'client'``: Uses a view that is purely client rendering (generally
+              safe without a virtual frame buffer)
+
+    name : str, optional
+        The name of the trame server on which the UI is defined
+
+    server_proxy_enabled : bool, default: False
+        Build a relative URL for use with ``jupyter-server-proxy``.
+
+    server_proxy_prefix : str, optional
+        URL prefix when using ``server_proxy_enabled``. This can be set globally in
+        the theme. To ignore, pass ``False``. For use with
+        ``jupyter-server-proxy``, often set to ``proxy/``.
+
+    collapse_menu : bool, default: False
+        Collapse the UI menu (camera controls, etc.) on start.
+
+    default_server_rendering : bool, default: True
+        Whether to use server-side or client-side rendering on-start when
+        using the ``'trame'`` mode.
+
+    return_viewer : bool, optional
+        Return the ipywidget.
+
+    **kwargs : dict, optional
+        Mostly ignored, though ``protocol`` and ``host`` can be use to
+        override the iframe src url and ``hieght`` and ``width`` can be
```

Comment:
```suggestion
        override the iframe src url and ``height`` and ``width`` can be
```
This also creates an issue. Before you could just set the width and height of the plotter, and the `ipyvtklink` ended up being the same size. So you should probably...